src/builder/definitions/base_class_def.py

```diff
@@ -1,16 +1,37 @@
 import inspect
 import typing
+from abc import abstractmethod
 
 from stubmaker.builder.common import Node, BaseRepresentationsTreeBuilder, BaseDefinition, get_annotations
 from typing_inspect import is_generic_type, get_generic_bases
 
 
+def _try_patch_pydantic_init_signature(cls):
+    """Pydantic implements custom class signature but assigns it only to cls.__signature__ but not
+    cls.__init__.__signature__ due to technical reasons: https://github.com/pydantic/pydantic/issues/1032. This
+    function tries to detect if the class is pydantic BaseModel and manually assigns signature to __init__.__signature__
+    """
+
+    try:
+        from pydantic import BaseModel
+        if issubclass(cls, BaseModel):
+            cls.__init__.__signature__ = cls.__signature__.replace(
+                parameters=[
+                    inspect.Parameter(name='self', kind=inspect.Parameter.POSITIONAL_OR_KEYWORD),
+                    *cls.__signature__.parameters.values(),
+                ],
+            )
+    except ImportError:
+        pass
```

Comment:
I don't think it is a good idea: the only case when it will be useful is when there are pydantic classes in the target code, but in this case this should never fail on import (it will fail during the import of user code). If there are no pydantic classes in user code this message will be annoying